types/submessages_test.go

```diff
@@ -0,0 +1,37 @@
+package types
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestReplySerialization(t *testing.T) {
+	// Deserializaton
+
+	// Serialization
+	reply1 := Reply{
+		GasUsed: 4312324,
+		ID:      75,
+		Result: SubMsgResult{
+			Ok: &SubMsgResponse{
+				Events: []Event{
```

Comment:
Some of these will change with the new `Array` type in #515, so I'll rebase that branch when this is merged.